reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -655,3 +614,42 @@ class StaticHtmlReporter : Reporter {
         unsafe { +renderer.render(document) }
     }
 }
+
+private fun ResolvedLicenseLocation.permalink(id: Identifier): String? {
+    val sourceArtifact = provenance.sourceArtifact
+    val vcsInfo = provenance.vcsInfo
+
+    return when {
+        sourceArtifact != null && sourceArtifact != RemoteArtifact.EMPTY -> {
+            val mavenCentralPattern = Regex("https?://repo[^/]+maven[^/]+org/.*")
+            when {
+                sourceArtifact.url.matches(mavenCentralPattern) -> {
+                    // At least for source artifacts on Maven Central, use the "proxy" from Sonatype which has the
+                    // Archive Browser plugin installed to link to the files with findings.
+                    with(id) {
+                        val group = namespace.replace('.', '/')
+                        "https://repository.sonatype.org/" +
+                                "service/local/repositories/central-proxy/" +
+                                "archive/$group/$name/$version/$name-$version-sources.jar/" +
+                                "!/${location.path}"
+                    }
+                }
+                else -> null
+            }
+        }
+
+        vcsInfo != null && vcsInfo != VcsInfo.EMPTY -> {
+            val path = listOfNotNull(
+                vcsInfo.path.takeIf { it.isNotEmpty() },
+                location.path
+            ).joinToString("/")
+
+            VcsHost.toPermalink(
+                vcsInfo.copy(path = path),
+                location.startLine, location.endLine
+            )
+        }
+
+        else -> null
```

Comment:
Can the else case actually happen? e.g. we could throw an exception instead then.

Reply:
It should not happen, but I don't want the reporter to fail just because it cannot generate the link.

Reply:
I think it's better to fail early here as it would be a bug and highlighting it would give us the opportunity to fix it easily, e.g. fail fast / early.

Reply:
From the perspective of the reporter this is not a bug but just insufficient information to generate a permalink. If we want to verify that the provenance fields cannot be empty this could be done in an evaluator rule, or in a dedicated place like the scanner, but not in a random place in one of the reporters.